### ISSUE TITLE
Additional auxiliary tasks

### ISSUE THREAD

**Issue by radi-cho** [bepoetree/MTTOD#10]
Congratulations on the good paper and code with their excellent results!
Can you share if during the research some other auxiliary tasks were tested before arriving at the one that is currently used and what were the outcomes? Also can you speculate on additional tasks that you think can achieve good results on the MultiWOZ dataset with T5?

**Comment by bepoetree** [comment]
Thank you for your interest :)

I have tested the span prediction task with various settings rather than testing other tasks.

For instance, I have adopted positional span prediction or BIO tag prediction instead of the token prediction.

But, these methods were less effective than token prediction.

There seems to be little additional tasks that do not require data annotation.

However, we are not utilizing all the information in MultiWOZ, such as slot description.

I think it would be helpful to use such information.

**Comment by bepoetree** [comment]
After we had corrected errors in evaluation code, a little performance degradation was observed.

please refer to issue https://github.com/bepoetree/MTTOD/issues/3.

**Comment by radi-cho** [comment]
Thank you. So if I write a paper and want to compare to your results, should I use `match: 92.30; success: 84.00; bleu: 19.41; score: 107.56` and state that these numbers are my own reproduction of your paper?

I also found a small inconsistency left in the code: The span head produces `num_span * 2 + 2` labels [here](https://github.com/bepoetree/MTTOD/blob/main/model.py#L204), but there are only `num_span + 2` span_tokens [here](https://github.com/bepoetree/MTTOD/blob/main/reader.py#L429). Which one is actually proven to work better?

**Comment by bepoetree** [comment]
1. Yes! You can use reproduction score or the [evaluation script](https://github.com/Tomiinek/MultiWOZ_Evaluation) which is used in the [leader board](https://github.com/budzianowski/multiwoz#response-generation).

2. I think it is a minor error. The number of span tokens is `num_span + 2`, but the output layer size is set by `num_span * 2 + 2`. 
It still works because the output layer size is larger than the vocabulary size. It is desirable to modify the [code](https://github.com/bepoetree/MTTOD/blob/main/model.py#L204) to `num_span + 2` (although it works the same way).

3. I have experimented on hyper-parameter settings such as the loss weights and learning rate rather than the optimization strategy. Thank you for sharing interesting results :)
